device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -106,12 +106,15 @@ public void onMessageSent(Message message, Throwable e)
         }
 
         // remove from in progress queue and add to callback queue
-        IotHubTransportPacket packet;
+        IotHubTransportPacket packet = null;
         synchronized (this.inProgressMessagesLock)
         {
-            //Codes_SRS_IOTHUBTRANSPORT_34_004: [This function shall retrieve a packet from the inProgressPackets
-            // queue with the message id from the provided message if there is one.]
-            packet = inProgressPackets.remove(message.getMessageId());
+            if (message != null)
```

Comment:
>null [](start = 27, length = 4)

add else statement on line 107 instead of checkout message != null in multiple places

Reply:
sure thing